hscontrol/types/node.go

```diff
@@ -97,6 +97,11 @@ type (
 	Nodes []*Node
 )
 
+// IsAutomaticNameMode returns whether the `givenName` can be automatically changed based on the `Hostname` of the node.
+func (node *Node) IsAutomaticNameMode() bool {
```

Comment:
I find this name quite confusing, maybe something like "GivenNameHasBeenChanged"?

Reply:
> I find this name quite confusing, maybe something like "GivenNameHasBeenChanged"?

It's not bad. I'll fix it now